external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java

```diff
@@ -134,22 +131,16 @@ public void execute(Tuple tuple) {
                     collector.ack(t);
                 tupleBatch.clear();
             }
+        } catch(SerializationError se) {
+            LOG.info("Serialization exception occurred, tuple is acknowledged but not written to Hive.", tuple);
+            collector.ack(tuple);
```

Comment:
Yeah, I think right try/catch blocks improve readability greatly but that
may be a style preference. Would a call to collector.reporterror also be
appropriate in that catch?

On Thu, Nov 12, 2015 at 8:44 AM Harsha <notifications@github.com> wrote:

> In
> external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java
> <https://github.com/apache/storm/pull/871#discussion_r44680811>:
>
> > @@ -134,22 +131,16 @@ public void execute(Tuple tuple) {
> >                      collector.ack(t);
> >                  tupleBatch.clear();
> >              }
> > +        } catch(SerializationError se) {
> > +            LOG.info("Serialization exception occurred, tuple is acknowledged but not written to Hive.", tuple);
> > +            collector.ack(tuple);
>
> @revans2 <https://github.com/revans2> makes sense. I'll add the metric
> and update the patch. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/storm/pull/871/files#r44680811>.
>